nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/CoreFeature.java

```diff
@@ -92,6 +93,40 @@
 @LocalDeploy("org.nuxeo.ecm.core.event:test-queuing.xml")
 public class CoreFeature extends SimpleFeature {
 
+    public class WorksWaiter implements Waiter {
+        @Override
+        public boolean await(long deadline) throws InterruptedException {
+            final WorkManager mgr = Framework.getService(WorkManager.class);
+            if (mgr.awaitCompletion(deadline - System.currentTimeMillis(), TimeUnit.MILLISECONDS)) {
+                return true;
+            }
+            logInfos(mgr);
+            return false;
+        }
+
+        protected void logInfos(final WorkManager mgr) {
+            String message = "Timed out while waiting for works";
+            StringBuffer bf = new StringBuffer();
+            bf.append(message + " ");
+            Iterator<String> ids = mgr.getWorkQueueIds().iterator();
+            while (ids.hasNext()) {
+                bf.append(System.lineSeparator());
+                String queueid = ids.next();
+                bf.append(mgr.getMetrics(queueid).toString());
+                bf.append(",works=");
+                Iterator<String> works = mgr.listWorkIds(queueid, null).iterator();
+                while (works.hasNext()) {
+                    bf.append(works.next());
+                    if (works.hasNext()) {
+                        bf.append(",");
+                    }
+                }
+            }
+            LogFactory.getLog(CoreFeature.class).error(bf.toString(), new Throwable("stack trace"));
```

Comment:
Use the `log` from the toplevel CoreFeature class. Make it package-private to avoid the synthetic accessor that would be needed if kept private. `toString` is not needed.